src/Common/src/System/Dynamic/Utils/DelegateHelpers.cs

```diff
@@ -59,8 +62,15 @@ private static Delegate CreateObjectArrayDelegateRefEmit(Type delegateType, Syst
             LocalBuilder retValue = ilgen.DeclareLocal(typeof(object));
 
             // create the argument array
-            ilgen.Emit(OpCodes.Ldc_I4, parameters.Length);
-            ilgen.Emit(OpCodes.Newarr, typeof(object));
+            if (parameters.Length == 0)
+            {
+                ilgen.Emit(OpCodes.Call, s_ArrayEmpty);
```

Comment:
Correct, we only emit the instructions against a volatile in-memory `DynamicMethod`.